cli/algorithms.go

```diff
@@ -3,15 +3,23 @@
 package cli
 
 import (
+	"crypto/ed25519"
 	"crypto/x509"
 	"encoding/pem"
+	"errors"
 	"log"
 	"os"
 
+	mgerr "github.com/absmach/magistrala/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/ultravioletrs/cocos/agent"
 )
 
+var (
+	errParseECP = errors.New("x509: failed to parse private key (use ParseECPrivateKey instead for this key format)")
```

Comment:
define a custom error, from this package

Reply:
and since all private keys implement the signer method, no need to switch between them, you can simply call the sign method